### PR TITLE
Add basic web socket coverage for Kestrel server

### PR DESCRIPTION
**Pull request by yuwaMSFT**
1. Add a simple MVC controller to handle web socket request
2. Add client script to run loadtest against the new controller using web socket

The scenario is very simple: the server reads input from client and sends back.
Since wcat doesn't support web socket, we use loadtest as the test client.

Usage:
startloadtest.cmd [Server:port] [NumConcurrentClients] [NumIterations]

or

startloadtest.sh [Server:port] [NumConcurrentClients] [NumIterations]

### CODE REVIEW

File: testapp/StressMvc/Controllers/WebSocketController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            
+            if (HttpContext.WebSockets.IsWebSocketRequest)
+            {
+                var webSocket = await HttpContext.WebSockets.AcceptWebSocketAsync();
+                await ProcessMessage(webSocket);
+            }
+            return new StatusCodeResult((int)HttpStatusCode.SwitchingProtocols);
+        }
+        private async Task ProcessMessage(WebSocket webSocket)
```
Comment (SajayAntony) [inline testapp/StressMvc/Controllers/WebSocketController.cs]: Newline.

File: testapp/StressMvc/Controllers/WebSocketController.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System.Net.WebSockets;
+using System.Net;
+using System.Threading;
```
Comment (SajayAntony) [inline testapp/StressMvc/Controllers/WebSocketController.cs]: Add license header?

File: testapp/StressMvc/Controllers/WebSocketController.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        private async Task ProcessMessage(WebSocket webSocket)
+        {
+
+            byte[] buffer = new byte[1024 * 4];
+            var result = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+            while (!result.CloseStatus.HasValue)
+            {
+                await webSocket.SendAsync(new ArraySegment<byte>(buffer, 0, result.Count), result.MessageType, result.EndOfMessage, CancellationToken.None);
+                result = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+            }
```
Comment (SajayAntony) [inline testapp/StressMvc/Controllers/WebSocketController.cs]: Add new lines.

File: testapp/StressMvc/appsettings.json [hunk 4]
```diff
@@ -4,6 +4,9 @@
       "ConnectionString": "Server=(localdb)\\mssqllocaldb;Database=aspnet5-StarterMvc-a41e748d-679d-48cc-a9f2-81b486f3512c;Trusted_Connection=True;MultipleActiveResultSets=true"
     }
   },
+  "WebSocket": {
```
Comment (SajayAntony) [inline testapp/StressMvc/appsettings.json]: Rename to `WebSocketOptions` to align with the property.

File: testapp/StressMvc/Startup.cs [hunk 5]
```diff
@@ -78,6 +78,10 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
 
             app.UseIdentity();
 
+            app.UseWebSockets(new WebSocketOptions
```
Comment (SajayAntony) [inline testapp/StressMvc/Startup.cs]: Please add a license header to the file.

**Comment by SajayAntony** [comment]
:watch:

**Comment by yuwaMSFT** [comment]
@SajayAntony Addressed the comments in latest commit.

**Comment by SajayAntony** [comment]
:shipit:

**Comment by SajayAntony** [comment]
:+1:

**Comment by yuwaMSFT** [comment]
manually merged to dev.

**Pull request closed without merging**